README.md

```diff
@@ -1,199 +1,240 @@
-            local opts = {
-                hook = {
-                    on_filetype = function()
-                        vim.api.nvim_buf_set_keymap(0, "n", "<Enter>", "<Plug>RDSendLine", {})
-                        vim.api.nvim_buf_set_keymap(0, "v", "<Enter>", "<Plug>RSendSelection", {})
-                    end
-                }
-                R_args = {"--quiet", "--no-save"},
-                min_editor_width = 72,
-                rconsole_width = 78,
-                disable_cmds = {
-                        "RClearConsole",
-                        "RCustomStart",
-                        "RSPlot",
-                        "RSaveClose",
-                    },
-                }
-                -- Check if the environment variable "R_AUTO_START" exists.
-                -- If using fish shell, you could put in your config.fish:
-                -- alias r "R_AUTO_START=true nvim"
-                if vim.env.R_AUTO_START == "true" then
-                    opts.auto_start = "on startup"
-                    opts.objbr_auto_start = true
-                end
-                require("r").setup(opts)
-            end,
-        lazy = false
+        "hrsh7th/nvim-cmp",
+        config = function()
+            require("cmp").setup({ sources = {{ name = "cmp_r" }}})
+            require("cmp_r").setup({})
+        end,
     },
+},
+```
+
+Note that [languageserver](https://github.com/REditorSupport/languageserver)
+can also be used for autocompletions, but using autocompletions from both
+sources simultaneously is not advised.
+
+### Tree-sitter
+
+Tree-sitter greatly enhances much of the functionality of R.nvim, and can be
+minimally configured like so:
+
+``` lua
+{
+    "nvim-treesitter/nvim-treesitter",
+    run = ":TSUpdate",
+    config = function ()
+        require("nvim-treesitter.configs").setup({
+            ensure_installed = { "markdown", "markdown_inline", "r", "rnoweb", "yaml" },
+            highlight = { enable = true },
+        })
+    end
+},
 ```
 
-The complete list of options is in the documentation.
-See also the [Wiki](https://github.com/R-nvim/R.nvim/wiki).
 
 ## Usage
 
-Please read the plugin's
-[documentation](https://github.com/R-nvim/R.nvim/blob/main/doc/R.nvim.txt) for
-instructions on usage. See also the output of `:RMapsDesc`.
+Please see the [documentation](doc/R.nvim.txt) for instructions on usage. For a
+complete list of keymaps, see the output of `:RMapsDesc`.
+
+## Lifecycle
+
+R.nvim is still maturing and its public API (configuration options,
+commands, and some of the Lua internals) may undergo breaking changes from
+time to time. This project uses [semantic versioning](https://semver.org/) to 
+help with this, and we will always bump the minor version, e.g. from 0.1.x to
+0.2.0, when we make a breaking change. Users are thus encouraged to pin their
+installation of R.nvim to the **latest minor release** and to check the release
+notes for any breaking changes when upgrading.
```

Comment:
Breaking changes should be when we change major, not minor right?

Reply:
Semver allows breaking changes before 1.0.0, but not after:

> 4. Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable.
>
> 5. Version 1.0.0 defines the public API. The way in which the version number is incremented after this release is dependent on this public API and how it changes.